addons/stock/models/stock_scrap.py

```diff
@@ -61,50 +53,43 @@ def _get_default_location_id(self):
         string='Status', default="draft", readonly=True, tracking=True)
     date_done = fields.Datetime('Date', readonly=True)
 
+    @api.depends('product_id')
+    def _compute_product_uom_id(self):
+        for scrap in self:
+            scrap.product_uom_id = scrap.product_id.uom_id
+
+    @api.depends('company_id', 'picking_id')
+    def _compute_location_id(self):
+        groups = self.env['stock.warehouse'].read_group(
+            [('company_id', 'in', self.company_id.ids)], ['min_id:min(id)'], ['company_id'])
+        locations_per_company = {
+            group['company_id'][0]: self.env['stock.warehouse'].browse(group['min_id']).lot_stock_id
+            for group in groups
+        }
+        for scrap in self:
+            if scrap.picking_id:
+                scrap.location_id = scrap.picking_id.location_dest_id if scrap.picking_id.state == 'done' else scrap.picking_id.location_id
```

Comment:
Why not leave this the way it was?

```suggestion
                scrap.location_id = (scrap.picking_id.state == 'done') and scrap.picking_id.location_dest_id or scrap.picking_id.location_id
```

Reply:
pylint says "consider using ternary'...